internal/service/cognitoidentity/pool_data_source.go

```diff
@@ -0,0 +1,198 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"allow_unauthenticated_identities": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"allow_classic_flow": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"openid_connect_provider_arns": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"saml_provider_arns": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"supported_login_providers": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"tags": tftags.TagsSchemaComputed(), // TIP: Many, but not all, data sources have `tags` attributes.
+		},
+	}
+}
+
+const (
+	DSNamePool         = "Pool Data Source"
+	ListPoolMaxResults = 20
+)
+
+func dataSourcePoolRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).CognitoIdentityConn(ctx)
+
+	name := d.Get("identity_pool_name").(string)
+
+	ip, err := findPoolByName(ctx, conn, name)
+	if err != nil {
+		return append(diags, create.DiagError(names.CognitoIdentity, create.ErrActionReading, DSNamePool, name, err)...)
+	}
+
+	d.SetId(aws.StringValue(ip.IdentityPoolId))
+
+	arn := arn.ARN{
+		Partition: meta.(*conns.AWSClient).Partition,
+		Region:    meta.(*conns.AWSClient).Region,
+		Service:   "cognito-identity",
+		AccountID: meta.(*conns.AWSClient).AccountID,
+		Resource:  fmt.Sprintf("identitypool/%s", d.Id()),
+	}
+	d.Set("arn", arn.String())
+	d.Set("identity_pool_name", ip.IdentityPoolName)
+	d.Set("allow_unauthenticated_identities", ip.AllowUnauthenticatedIdentities)
+	d.Set("allow_classic_flow", ip.AllowClassicFlow)
+	d.Set("developer_provider_name", ip.DeveloperProviderName)
+
+	setTagsOut(ctx, ip.IdentityPoolTags)
+
+	if err := d.Set("cognito_identity_providers", flattenIdentityProviders(ip.CognitoIdentityProviders)); err != nil {
+		return sdkdiag.AppendErrorf(diags, "setting cognito_identity_providers error: %s", err)
+	}
+
+	if err := d.Set("openid_connect_provider_arns", flex.FlattenStringList(ip.OpenIdConnectProviderARNs)); err != nil {
+		return sdkdiag.AppendErrorf(diags, "setting openid_connect_provider_arns error: %s", err)
+	}
+
+	if err := d.Set("saml_provider_arns", flex.FlattenStringList(ip.SamlProviderARNs)); err != nil {
+		return sdkdiag.AppendErrorf(diags, "setting saml_provider_arns error: %s", err)
+	}
+
+	if err := d.Set("supported_login_providers", aws.StringValueMap(ip.SupportedLoginProviders)); err != nil {
+		return sdkdiag.AppendErrorf(diags, "setting supported_login_providers error: %s", err)
+	}
+
+	return diags
+}
+
+func findPoolByName(ctx context.Context, conn *cognitoidentity.CognitoIdentity, name string) (*cognitoidentity.IdentityPool, error) {
```

Comment:
can this be reused for the resource?

Reply:
With the resource they already have the ID so they don't need to search by name. The reason for this function is I have to List until I find a match to get the ID